File: problems/002-two-sum/php/brute-force.php
<?php
// Brute force: check every pair. The "obvious" first approach --
// O(n^2) time, O(1) space.
function solve(array $c): array {
    $nums = $c['nums'];
    $n = count($nums);
    for ($i = 0; $i < $n; $i++) {
        for ($j = $i + 1; $j < $n; $j++) {
            if ($nums[$i] + $nums[$j] === $c['target']) return [$i, $j];
        }
    }
    return [];
}

File: problems/002-two-sum/csharp/Brute-force.cs
using System.Linq;
using System.Collections.Generic;
using System.Text.Json;
class BruteForce : ISolution {
    public object Solve(Dictionary<string, object> c) {
        long target = ((JsonElement)c["target"]).GetInt64();
        var nums = ((JsonElement)c["nums"]).EnumerateArray().Select(e => e.GetInt64()).ToArray();
        // Check every pair -- the obvious first approach, O(n^2).
        for (int i = 0; i < nums.Length; i++)
            for (int j = i + 1; j < nums.Length; j++)
                if (nums[i] + nums[j] == target) return new[] { i, j };
        return new int[0];
    }
}
